### ISSUE TITLE
allow cloning a TypedDict via dict comprehension

### ISSUE THREAD

**Issue by danielhollas** [astral-sh/ty#3942]
### Question

It would be nice if this kind of code worked:

```python
from typing import reveal_type, TypedDict

class A(TypedDict, closed=True):
    x: list[int]
    y: list[str]

d = A(x=[1], y=['a'])
d_copy: A = {key: val.copy() for key, val in d.items()}
```
This now results in 
```
Object of type `dict[str, list[int] | list[str]]` is not assignable to `A` 
```

https://play.ty.dev/1db79d31-0c4c-4b01-aecd-2a65c9237377

Perhaps related to #3563?

Also happy to take suggestion to how to write this without triggering false positive in current ty version.

### Version

ty 0.0.57

EDIT: Fixed the link to playground and the error

**Comment by carljm** [comment]
This would be cool. I think it would be quite challenging/impossible to do it via composing fully general type system features, so it would likely require fairly special-cased recognition of this particular "comprehension over a TypedDict" pattern. It's unclear how broadly impactful that would be.

One initial step (which is fairly general) towards this would be to fix this TODO: https://github.com/astral-sh/ruff/blob/8593e828341befc04f4e43119d69c9ca11e4ad7b/crates/ty_python_semantic/resources/mdtest/typed_dict.md#L5736-L5738

But even then, I think it would require narrow special-casing of the overall pattern to get around the fact that unpacking to `key, val` in general has to collapse those to independent union types, losing the correlation between specific key type and specific value type.
